presto-spi/src/main/java/com/facebook/presto/spi/function/RoutineCharacteristics.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.function;
+
+import java.util.Objects;
+
+import static com.facebook.presto.spi.function.RoutineCharacteristics.Determinism.DETERMINISTIC;
+import static com.facebook.presto.spi.function.RoutineCharacteristics.NullCallClause.CALLED_ON_NULL_INPUT;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class RoutineCharacteristics
```

Comment:
If you really want to introduce this class, make this part of `SqlFunction`.

Reply:
determinism and call convention are needed for all queries, and they are boolean. They already exist in SqlFunction. I still don't think it's necessary to introduce this class. They can be rolled into the SqlFunction / SqlInvokedRegularFunction directly.